src/alks.ts

```diff
@@ -664,15 +689,24 @@ namespace ALKS {
      * })
      */
     async createRole(props: CreateRoleProps): Promise<Role> {
-      const results = await this.internalFetch('createRole', props);
-      results.denyArns = results.denyArns.split(',');
-      return pick(results, [
-        'roleArn',
-        'denyArns',
-        'instanceProfileArn',
-        'addedRoleToInstanceProfile',
-        'tags',
-      ]);
+      let roleTypeExists = props.roleType != null;
+      let trustPolicyExists = props.trustPolicy != null;
+      if (
+        (roleTypeExists && !trustPolicyExists) ||
+        (!roleTypeExists && trustPolicyExists)
+      ) {
```

Comment:
Spencer had a good idea in the ALKS Core repo that I'd like to replicate here. He basically used the XOR operator `^` (this works the same in JS as it does in Java) which is true if one of the operands is true, but false if either both are true or both are false. So you can simplify this boolean expression to this:

```suggestion
      if (!(roleTypeExists ^ trustPolicyExists)) {
```

Technically `^` is a bitwise operator which will coerce the types for `roleTypeExists` and `trustPolicyExists` into byte values, but the `!` operator will coerce the result of the operation back into the correct boolean value so it all works out fine